### PR TITLE
Scaffolding

### PR DESCRIPTION
**Pull request by mj850**
## Describe your changes and provide context
Adds the basic scaffolding for the rest of the app

### CODE REVIEW

File: x/evm/types/params.pb.go [hunk 1]
```diff
@@ -104,53 +104,122 @@ func (m *Params) GetTargetGasUsedPerBlock() uint64 {
 	return 0
 }
 
+type ParamsPreV580 struct {
```
Comment (dssei) [inline x/evm/types/params.pb.go]: not sure why that is being generated. Seen this too, but dropped

File: proto/confidentialtransfers/confidential.proto [hunk 2]
```diff
@@ -0,0 +1,18 @@
+syntax = "proto3";
+package seiprotocol.seichain.confidentialtransfers;
+
+import "gogoproto/gogo.proto";
+import "confidentialtransfers/cryptography.proto";
+
+option go_package = "github.com/sei-protocol/sei-chain/x/confidentialtransfers/types";
+
+
+// CtAccount represents an confidential account state
```
Comment (dssei) [inline proto/confidentialtransfers/confidential.proto]: nit: a confidential transfers account state

File: proto/confidentialtransfers/params.proto [hunk 3]
```diff
@@ -0,0 +1,7 @@
+syntax = "proto3";
+package seiprotocol.seichain.confidentialtransfers;
+
+option go_package = "github.com/sei-protocol/sei-chain/x/confidentialtransfers/types";
+
+// Params defines the parameters for the tokenfactory module.
```
Comment (dssei) [inline proto/confidentialtransfers/params.proto]: Typo: tokenfactory

File: x/confidentialtransfers/types/account.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+	// This is calculated as Encrypted(encryptionPK, <available_balance>)
+	AvailableBalance *elgamal.Ciphertext
+
+	// The Asymmetrically Encrypted available balance.
+	// This is calculated as AsymmetricEncryption(otherPK, <available_balance>)
+	// This is stored as the Base64-encoded ciphertext
+	DecryptableAvailableBalance string
+}
+
+func (m *Account) ToProto() *CtAccount {
```
Comment (dssei) [inline x/confidentialtransfers/types/account.go]: For consistency, either my changes have to move to `ToProto` to non proto objects or this has method would move to `CtAccount`

File: x/confidentialtransfers/types/ct_account.go [hunk 5]
```diff
@@ -0,0 +1,79 @@
+package types
+
+import (
+	"github.com/coinbase/kryptology/pkg/core/curves"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/sei-protocol/sei-cryptography/pkg/encryption/elgamal"
+)
+
+func (m *CtAccount) ValidateBasic() error {
+	account, err := m.FromProto()
```
Comment (dssei) [inline x/confidentialtransfers/types/ct_account.go]: I have moved away from converting stuff in ValidateBasic btw, just checking if fields are there.

File: x/confidentialtransfers/types/account.go (hunk 4 above)
Comment (mj850) [reply x/confidentialtransfers/types/account.go]: I changed this to a constructor in CtAccount instead - (NewCtAccount).

**Pull request merged**